src/main/java/fi/otavanopisto/kuntaapi/server/id/AbstractIdFactory.java

```diff
@@ -0,0 +1,61 @@
+package fi.otavanopisto.kuntaapi.server.id;
+
+import fi.otavanopisto.kuntaapi.server.integrations.KuntaApiConsts;
+import fi.otavanopisto.kuntaapi.server.persistence.model.Identifier;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.inject.Inject;
+
+public abstract class AbstractIdFactory {
+
+  @Inject
+  private Logger logger;
+  
+  public <T extends BaseId> T createKuntaApiId(Class<T> idClass, Identifier identifier) {
+    if(identifier.getOrganizationKuntaApiId() == null) {
+      return createId(idClass, null, KuntaApiConsts.IDENTIFIER_NAME, identifier.getKuntaApiId());
+    } else {
+      OrganizationId organizationId = new OrganizationId(KuntaApiConsts.IDENTIFIER_NAME, identifier.getOrganizationKuntaApiId());
+      return createId(idClass, organizationId, KuntaApiConsts.IDENTIFIER_NAME, identifier.getKuntaApiId());
+    }
+  }
+  
+  protected <T extends BaseId> T createId(Class<T> idClass, OrganizationId organizationId, String source, String id) {
+    if(isOrganizationBaseId(idClass)) {
+      return createOrganizationBaseId(idClass, source, id, organizationId);
+    } else {
+      return createBaseId(idClass, source, id);
+    }
+  }
+  
+  private boolean isOrganizationBaseId(Class<? extends BaseId> idClass) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "isOrganizationBaseId" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)